run/pubsub/test/app.test.js

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019, Google LLC.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// NOTE:
+// This app can only be fully tested when deployed, because
+// Pub/Sub requires a live endpoint URL to hit. Nevertheless,
+// these tests mock it and partially test it locally.
+
+'use strict';
+
+const assert = require('assert');
+const path = require('path');
+const tools = require('@google-cloud/nodejs-repo-tools');
+const uuid = require('uuid');
+
+const cwd = path.join(__dirname, '../');
+const requestObj = tools.getRequest({cwd});
```

Comment:
@fhinkel this uses `nodejs-repo-tools` - I assume that's OK for this sample, since it's not a simple hello-world?

Reply:
We're trying to remove the usage of `repo-tools`. Can you use `request` or `fetch` instead of `tools.getRequest`?

Reply:
You can have a look at the appengine Hello World example, we use `supertest` for this kind of testing. Since we're not running e2e tests there's no need to install `repo-tools`. But not blocking, I'll work on removing `repo-tools` from the samples anyways.

Reply:
Switched to sinon and supertest.